core/src/test/java/org/infinispan/util/BaseControlledConsistentHashFactory.java

```diff
@@ -19,27 +22,38 @@
  * @since 6.0
  */
 @SuppressWarnings("unchecked")
-public abstract class BaseControlledConsistentHashFactory implements ConsistentHashFactory<DefaultConsistentHash>,
+public abstract class BaseControlledConsistentHashFactory<CH extends ConsistentHash> implements ConsistentHashFactory<CH>,
```

Comment:
What is the advantage of using generics here? Couldn't we create a standard inheritance here? Also since this is an abstract class, maybe we could call it `AbstractControlledConsistentHashFactory`?